app/src/main/java/com/example/bigowlapp/utils/PermissionsHelper.java

```diff
@@ -0,0 +1,175 @@
+package com.example.bigowlapp.utils;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.widget.Toast;
+
+import androidx.annotation.RequiresApi;
+import androidx.appcompat.app.AlertDialog;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+
+import com.example.bigowlapp.R;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This helper can be used when needing to handle calls for permissions
+ * without bloating an Android UI class, and to reduced repeated code.
+ */
+public class PermissionsHelper {
```

Comment:
Really cool class to handle permissions.

Though, I have a question as I don't really understand some of the code. Are most of these methods flexible? As in, if I'm requesting for another permission other than location, will this class help?

Reply:
Yep, I created some methods for location specifically as background location has it's own complexity, but everything else can use `requestMissingPermissions()`.

Anything public in `PermissionsHelper` can be relevant for anyone to use.

If there's a public one that shouldn't be public, just tell me.